File: sol/solution/0001-0100/0003. Longest Substring Without Repeating Characters/Solution.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<char, int> last; 
        int res = 0, start = 0;
        for(int i = 0; i < int(s.size()); ++i) {
            if (last.count(s[i]))
                start = max(start, last[s[i]]);
            res = max(res, i - start + 1);
            last[s[i]] = i + 1;
        }
        return res;
    }
};

File: sol/solution/0001-0100/0003. Longest Substring Without Repeating Characters/Solution.swift
class Solution {
    func lengthOfLongestSubstring(_ s: String) -> Int {
        var last = [Character: Int]()
        var res = 0, start = 0
        for (i, c) in s.enumerated() {
            if let idx = last[c], idx >= start {
                start = idx + 1
            }
            res = max(res, i - start + 1)
            last[c] = i
        }
        return res
    }
}